blog-content/articles/testing-makes-you-faster-day-one.md

```diff
@@ -155,5 +160,83 @@ class ChestTest {
+
+# Real Life Boring Example
+
+You probably don't face castles and dragons every day at your work, so let me map it to a boring real life example.
+
+Let's add a page to Enterprise Corporate Website. Page will list employees that have birthdays today. List must be in an alphabetic order.
+
+Your fire up your IDE, write new page, deploy to DEV environment and test it. All looks good.
+A minute later you realize that your company doesn't have employees with the same last name that have birthdays on the sae day. 
+So how do you test this scenario?
+
+You create a SysAdmin Ticket to request `WRITE` access to `DEV` database. Wait for approval. Restart the server and check your code.
+
+Note: You may say, but hey, I can run my database locally! Sure, but not every big company and not every software has moved on to the greatest tech.
+And most excuses for not writing tests come from Enterprise engineers working on legacy applications. Plus it helps to make a point.   
+
+Your code is in production for a month. Everyone is happy, the page is a great success and have many daily visitors. 
+Hence a new request comes up: you need to change the sorting works.
+
+````columns
+left:
+```text {title: "insted of"}
+Smith Bo
+Smith Bob
+Smith Bobed
+Smith Al
+Smith Alice
+```
+
+right:
+```text {title: "expected"}
+Smith Alice
+Smith Al
+Smith Bobed
+Smith Bob
+Smith Bo
+```    
+````
+
+Basically, still sort alphabetically, but put longer names at the top. It makes page looks better!
+
+You fire up your `DEV` to realize the data is gone. You create a SysAdmin Ticket to request `WRITE` access to `DEV` database. 
+Wait for approval. Restart the server and start coding. 
+
+Let's rewind and try again. Instead of jumping to deploy and database manipulations, let's build a teleport.
+
+```java {title: "Employees Sort Teleport"}
+class EmployeesSorterTest {
+    @Test
+    public void matchingLastName() {
+        EmployeesList list = new EmployeesList();
+        list.add("Smith", "Bobed");
+        list.add("Smith", "Alice");
+        list.add("Smith", "Al");
+        list.add("Smith", "Bob");
+
+        EmployeesSorter employeesSorter = new EmployeesSorter();
+        EmployeesList sorted = employeesSorter.sort(list);
+        
+        Assert.assertEqual("Alice", sorted.get(0).getFirstName());
+        Assert.assertEqual("Al", sorted.get(1).getFirstName());
+        Assert.assertEqual("Bobed", sorted.get(2).getFirstName());
+        Assert.assertEqual("Bob", sorted.get(3).getFirstName());
+    }
+}
+``` 
+
+Whether you implement `EmployeesSorterTest` first or `EmployeesSorter` is not that important to me.
+
+What important is you have a very tight feedback loop between your changes and the result. 
+You bypass database admins (ahem, dragon, ahem, JK), web pages, internal auth systems and get straight to the business logic at hand.
+
+# Summary
```

Comment:
I really love the summary. This article does the following 3 things very well: 
1. It tells the reader what they are going to read
2. The thing
3. It tells the reader what they just read

Adding #3 by creating a summary just supercharged the article :)

Reply:
you are too kind, sir